### DISCUSSION TITLE
Order of Flows

### DISCUSSION THREAD

**Discussion by jakedowns** [directus/directus#16330] [Archive]
is there a way to control the order of flows? do they run in alphabetical order, or like, just first created first executed?

cause like, if i have multiple item.create scoped/triggered flows, how do i know which will execute first?

what happens if there are multiple Blocking filters? 

what if i want an Action to complete, while also having an active Blocking filter?

**Comment by jakedowns** [comment]
also, am i crazy or is this a known issue? https://github.com/directus/directus/issues/16329

**Comment by br41nslug** [comment]
> do they run in alphabetical order, or like, just first created first executed?

Flows get ran in the order they're fetched which might be a tad random as by default the will sort by the UUID primary key.
> if i have multiple item.create scoped/triggered flows, how do i know which will execute first?

You don't really know without fetching the flows manually and checking the order. Flows however are intended to be isolated pieces of logic if you have 2 flows on the same event where they're dependent on each other it might be worth it considering merging that logic into one flow (or chaining them after another in the preferred order if they're separated for a reason).
> what happens if there are multiple Blocking filters?

The same except the flows will wait for the previous one to finish
> what if i want an Action to complete, while also having an active Blocking filter?

This is not possible filters need to complete before an action can be triggered.
